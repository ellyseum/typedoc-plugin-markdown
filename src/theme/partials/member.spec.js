describe(`member.hbs`, () => {
  test('should compile with delcaration', () => {
    const context = { name: 'member' };
    const result = global.compileTemplate('member.hbs', context);
    expect(result).toMatchSnapshot();
  });

  test('should compile with signature', () => {
    const context = {
      name: 'member',
      signatures: [{}],
    };
    const result = global.compileTemplate('member.hbs', context);
    expect(result).toMatchSnapshot();
  });

  test('should compile with getter setter', () => {
    const context = {
      name: 'member',
      hasGetterOrSetter: () => true,
    };
    const result = global.compileTemplate('member.hbs', context);
    expect(result).toMatchSnapshot();
  });
});
