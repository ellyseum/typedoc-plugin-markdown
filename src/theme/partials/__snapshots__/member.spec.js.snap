// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`member.hbs should compile with delcaration 1`] = `
"[member.title]

[member.declaration]

___"
`;

exports[`member.hbs should compile with getter setter 1`] = `
"[member.title]

[member.getterSetter]

___"
`;

exports[`member.hbs should compile with signature 1`] = `
"[member.title]

[member.signatures]

___"
`;
